src/pages/home/report/ReportActionsView.js

```diff
@@ -151,33 +152,21 @@ class ReportActionsView extends React.Component {
      */
     onVisibilityChange() {
         if (Visibility.isVisible()) {
-            this.timers.push(setTimeout(this.recordMaxAction, 3000));
+            this.unreadTimer = setTimeout(this.scheduledRecordMaxAction, 3000);
         }
     }
 
     /**
-     * Checks if the unreadActionIndicator should be shown.
-     * If it does, starts a timeout for the fading out animation and creates
-     * a flag to not show it again if the report is still open
+     * Helper function to clean up the timer before setting the max action
      */
-    setUpUnreadActionIndicator() {
-        if (!this.shouldShowUnreadActionIndicator) {
-            return;
-        }
-
-        this.unreadActionCount = this.props.report.unreadActionCount;
-
-        if (this.unreadActionCount > 0) {
-            this.unreadIndicatorOpacity = new Animated.Value(1);
-            this.timers.push(setTimeout(() => {
-                Animated.timing(this.unreadIndicatorOpacity, {
-                    toValue: 0,
-                    useNativeDriver: false,
-                }).start();
-            }, 3000));
+    scheduledRecordMaxAction() {
+        // Always cancel the existing timer
```

Comment:
What happens if we do not cancel this timer ?

Reply:
Related to the answer above: I got rid of it so I could check for existence. It may no longer be needed.